### PR TITLE
(Support) Expose onTabChanged

### PR DESCRIPTION
**Pull request by akwanpdf**
Exposed `onTabChanged` event for Android and iOS.

### CODE REVIEW

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 1]
```diff
@@ -2410,6 +2414,102 @@ public void onStateChanged() {
         }
     };
 
+//    private PdfViewCtrlTabHostFragment2.TabHostListener mTabHostListener = new PdfViewCtrlTabHostFragment2.TabHostListener() {
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:2417]: ?

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 2]
```diff
@@ -2646,6 +2747,12 @@ public void onTabChanged(String tag) {
         super.onTabChanged(tag);
         if (getPdfViewCtrlTabFragment() != null) {
             getPdfViewCtrlTabFragment().setStateEnabled(mSaveStateEnabled);
+
+            WritableMap params = Arguments.createMap();
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:2751]: can be outside of if statement

File: API.md [hunk 3]
```diff
@@ -1034,11 +1034,34 @@ Sets the limit on the maximum number of tabs that the viewer could have at a tim
...
+
+The function is activated when a tab is changed. 
+
+Please note that this API is meant for tab-specific changes. If you would like to know when the document finishes loading instead, see the [`onDocumentLoaded`](#onDocumentLoaded) event.
+
+Parameters:
+
+Name | Type | Description
+--- | --- | ---
+currentTab | string | The title of the current tab
```
Comment (sgong-pdftron) [inline API.md:1053]: on android it's the full path i believe

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java (hunk 1 above)
Comment (dcupidon) [reply android/src/main/java/com/pdftron/reactnative/views/DocumentView.java]: Removed the commented out code

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java (hunk 2 above)
Comment (dcupidon) [reply android/src/main/java/com/pdftron/reactnative/views/DocumentView.java]: Moved the code outside the if statement

File: API.md (hunk 3 above)
Comment (dcupidon) [reply API.md]: On both iOS and Android the full path is returned, so the documentation has been updated

**Review by sgong-pdftron — APPROVED** [review]

File: ios/RNTPTDocumentView.m [hunk 4]
```diff
@@ -2767,6 +2780,21 @@ - (BOOL)tabbedDocumentViewController:(PTTabbedDocumentViewController *)tabbedDoc
...
+    if (context == TabChangedContext) {
+        if ([self.delegate respondsToSelector:@selector(tabChanged:currentTab:)]) {
+            PTDocumentTabItem *selectedItem = self.tabbedDocumentViewController.tabManager.selectedItem;
+            if (selectedItem != nil) {
+                NSURL *currentTab = selectedItem.documentURL ?: selectedItem.sourceURL;
+                [self.delegate tabChanged:self currentTab:[currentTab absoluteString]];
+            }
+        }
+    } else {
+        [super observeValueForKeyPath:keyPath ofObject:object change:change context:context];
```
Comment (JamesTron) [inline ios/RNTPTDocumentView.m:2794]: You should still call this even when you handle it. So you can call this blindly at the beginning of the method, then only do your special stuff if `context == TabChangedContext`.

**Review by JamesTron — CHANGES_REQUESTED** [review]

File: ios/RNTPTDocumentView.m (hunk 4 above)
Comment (akwanpdf) [reply ios/RNTPTDocumentView.m:2794]: (Resolving as no longer necessary)

**Pull request merged**